### PR TITLE
Add variable width spaghetti plot

### LINKED ISSUE #807: Feature request: Option to vary qwidth in spaghetti_plot
In `spaghetti_plot` at the moment the integration range over the perpendicular directions for each q-line segment is fixed to a single value in reciprocal angstrom (default is `0.1`). This ensures that the slices are all of uniform width but is a bit restrictive for users.

So, some users have requested the ability to specify different widths for each segment. The use case for this would be to increase or decrease the widths in directions where the excitations are (relatively) dispersionless, for example in low-dimensional magnets, in order to improve the (effective) resolution [with smaller widths] or the data quality [larger widths].

### PR DESCRIPTION
**Pull request by oerc0122**
- Add variable widths for q-segments to spaghetti plot
- Make noplot live up to its name (even in case of no action)
- Correct invalid_arguments to invalid_argument

Fixes #807

### CODE REVIEW

File: horace_core/spaghetti_plot.m [hunk 1]
```diff
@@ -362,7 +379,7 @@
 
 % Plot dispersion
 %----------------
-if nargout<1 || (~present.noplot)
+if nargout<1
     plot_dispersion(arrayfun(@d2d, wdisp),opt)
 end
```
Comment (abuts) [inline horace_core/spaghetti_plot.m:384]: If I correctly understand the change, I would disagree with this. 
You can request plot and ask to return objects -- pretty reasonable situation. Here you allow one or anoter

File: horace_core/spaghetti_plot.m [hunk 2]
```diff
@@ -139,7 +144,7 @@
...
 
 if numel(args) ~= 2
-    error('SPAGHETTI_PLOT:invalid_arguments', ...
+    error('SPAGHETTI_PLOT:invalid_argument', ...
           'Invalid number of arguments')
 end
```
Comment (abuts) [inline horace_core/spaghetti_plot.m:149]: I believe, we have agreement to throw 3-component exceptions. Standard exception would be useful if you, e.g. write GUI and catch all normal exceptions to display them in different place in GUI. 

I do not mind if it is not the case here, but having standard 3-component exception 'HORACE:spaghetti_plot:invalid_argument' would simplify further development

**Review by abuts — CHANGES_REQUESTED** [review]

File: horace_core/spaghetti_plot.m (hunk 2 above)
Comment (oerc0122) [reply horace_core/spaghetti_plot.m:149]: Agreed, fixed.

File: horace_core/spaghetti_plot.m (hunk 1 above)
Comment (oerc0122) [reply horace_core/spaghetti_plot.m]: The change in this PR does not affect that at all, this is the original behaviour. 

The idea, I believe, is that if you return the object you clearly have an intent to transform it or use it elsewhere and so don't need to plot it immediately. 

This is indeed an overloaded use of the function and I am not a fan of that in general, but I recommend keeping the behaviour in this PR and if you want to discuss changing it, I would say open an issue.

**Review by abuts — APPROVED** [review]

**Review by cmarooney-stfc — APPROVED** [review]

**Pull request merged**